lib/msal-common/docs/multi-tenant-accounts.md

```diff
@@ -0,0 +1,184 @@
+
+```javascript
+const msalConfig = {
+    auth: {
+        clientId: "ENTER_CLIENT_ID",
+        authority: "https://login.microsoftonline.com/HOME_TENANT", // This is the authority that MSAL will default to for requests that don't specify their own authority.
+    },
+};
+
+const myMSALObj = new PublicClientApplication(msalConfig);
+
+// handleRedirectPromise has to be called to resume redirect requests
+myMSALObj
+    .handleRedirectPromise()
+    .then(handleResponse)
+    .catch((err) => {
+        console.error(err);
+    });
+
+// Authority isn't specified because it is already the default set in the MSAL Config object
+const homeTenantRequest = {
+    scopes: ["HOME_TENANT_SCOPE"],
+};
+
+// Guest tenant authority overrides the MSAL config authority, meaning the user can authenticate with the tenant they are a guest in and acquire tokens for it
+const guestTenantRequest = {
+    scopes: ["GUEST_TENANT_SCOPE"],
+    authority: "https://login.microsoftonline.com/GUEST_TENANT",
+};
+
+const homeTenantAuthResponse = await myMSALObj.loginPopup(homeTenantRequest);
+// Get the home tenant account from the AuthenticationResult
+const homeTenantAccount = homeTenantAuthResponse.account;
+
+const guestTenantAuthResponse = await myMSALObj.loginPopup(homeTenantRequest);
+// Get the guest tenant account from the AuthenticationResult
+const guestTenantAccount = guestTenantAuthResponse.account;
+```
+
+Assuming the user logs into both tenants with the same account, once both login requests are successfully completed and the user has authenticated with each tenant, the cache will contain:
+
+-   2 ID tokens (one per tenant)
+-   2 Access tokens (one per tenant)
+-   1 Refresh token (one per account, shared across tenants)
+-   1 AccountEntity containing both tenant profiles. The `homeTenantAccount` and `guestTenantAccount` objects shown above are both created from the same cached account entity augmented by the claims in their respective ID tokens.
+
+### Acquiring Tokens Silently for Multiple Tenants
+
+In order to acquire cached tokens for a particular tenant, the `AccountInfo` object for that particular tenant must be passed into `acquireTokenSilent`. In order to get the correct account to pass into acquire token silent, you can either:
+
+-   Use the `AccountInfo` object returned from an interactive API call to that tenant:
+
+```javascript
+// Original interactive call to home tenant
+const homeTenantAuthResponse = await myMSALObj.loginPopup(homeTenantRequest);
+// Get the home tenant account from the AuthenticationResult
+const homeTenantAccount = homeTenantAuthResponse.account;
+.
+.
+.
+// Acquire a home tenant access token silently
+const homeTenantSilentResponse = await myMSALObj.acquireTokenSilent({
+    account: homeTenantAccount,
+    ...homeTenantRequest,
+});
+
+
+// Original interactive call to guest tenant
+const guestTenantAuthResponse = await myMSALObj.loginPopup(homeTenantRequest);
```

Comment:
Do you actually need an interactive call for each separate tenant or can you go straight to ATS since the RTs are shared?

Reply:
Assuming this was needed due to not having the accountinfo object yet for the tenant you're requesting. I'd suggest instead that if you don't have the tenanted account yet folks can invoke ATS with the home account, tenanted authority & forceRefresh or CachePolicy to silently get the guest tenant tokens

Reply:
Reworked the sample code to handle this better using a custom function that checks if the tenant profile is already in the cache and if not it uses acquireTokenSilent with the `CacheLookupPolicy.RefreshToken` with the guest tenant authority.

FWIW, `ssoSilent` can also be used but I didn't want to add complexity since the doc is already long and verbose and that's msal-browser specific. I'll add a section to the login docs or a special doc for it soon.